Framework/CurveFitting/test/Functions/AsymmetricPearsonVIITest.h

```diff
@@ -0,0 +1,596 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include <cxxtest/TestSuite.h>
+#include <math.h>
+#include <memory>
+
+#include "MantidAPI/PeakFunctionIntegrator.h"
+#include "MantidCurveFitting/CostFunctions/CostFuncLeastSquares.h"
+#include "MantidCurveFitting/FuncMinimizers/LevenbergMarquardtMDMinimizer.h"
+#include "MantidCurveFitting/Functions/AsymmetricPearsonVII.h"
+#include "MantidCurveFitting/Functions/Gaussian.h"
+#include "MantidCurveFitting/Functions/Lorentzian.h"
+#include "MantidCurveFitting/Jacobian.h"
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::CurveFitting;
+using namespace Mantid::CurveFitting::Functions;
+using namespace Mantid::CurveFitting::CostFunctions;
+
+class AsymmetricPearsonVIITest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static AsymmetricPearsonVIITest *createSuite() { return new AsymmetricPearsonVIITest(); }
+  static void destroySuite(AsymmetricPearsonVIITest *suite) { delete suite; }
+
+  AsymmetricPearsonVIITest() : m_xValues(), m_10xValues() {
+    for (size_t i = 0; i < 2000; ++i) {
+      double x_i = -1 + 0.001 * static_cast<double>(i);
+      m_xValues.emplace_back(x_i);
+    }
+
+    for (size_t i = 0; i < 10; ++i) {
+      double x_i = -1 + 0.2 * static_cast<double>(i);
+      m_10xValues.emplace_back(x_i);
+    }
+  }
+
+  void testCategory() {
+    AsymmetricPearsonVII ap7;
+    TS_ASSERT_EQUALS(ap7.category(), "XrayDiffraction");
+  }
+
+  void testSetParameters() {
+    AsymmetricPearsonVII ap7;
+    const double ph = 0.9;
+    const double pc = 0.5;
+    const double width = 1.1;
+    const double ml = 2.0;
+    const double mr = 0.0;
+    ap7.initialize();
+    ap7.setHeight(ph);
```

Comment:
before we set the new values check the default values. This then confirms that you have actually changed them